### ISSUE TITLE
Question about eesen training code

### ISSUE THREAD

**Issue by SFuji** [srvk/eesen#29]
Hi ,

There is a question for me while reading the eesen code. 
Location: line 66 to line 75 at file net/ctc_loss.cc
When back-propagate the errors through the softmax layer, as I can get from the code, 
the formula is ctc_error \* yk - Row_mul(yk \* ColSum(ctc_error \* yk) ).
But the formula of softmax-derivation is yk \* (1 - yk).
So as I can get, the difference of using softmax-derivation formular is the ColSum and RowMul.
And why? Is there something I missed?

Looking forward for reply!

**Comment by yajiemiao** [comment]
![formula](https://cloud.githubusercontent.com/assets/15018842/12423008/9328a516-be97-11e5-9af5-95cd1ad3e59c.png)

Here are the derivation for the gradients, from after-softmax to before-softmax

**Comment by yajiemiao** [comment]
lines 66~75 of net/ctc_loss.cc exactly follow Equation 8(c) in the attached image.

**Comment by SFuji** [comment]
I have got the point. Thanks.
